addons/l10n_tr_nilvera_einvoice/models/account_edi_xml_ubl_tr.py

```diff
@@ -42,8 +44,19 @@ def _get_formatted_id(invoice):
             'pricing_currency_code': invoice.currency_id.name.upper() if invoice.currency_id != invoice.company_id.currency_id else False,
             'currency_dp': 2,
         })
+        vals['vals']['note_vals'].append({'note': self._get_amount_in_text(invoice.amount_residual_signed, self.env.ref('base.TRY')), 'note_attrs': {}})
+        if vals['invoice'].currency_id.name != 'TRY':
+            vals['vals']['note_vals'].append({'note': self._get_amount_in_text(invoice.amount_residual, vals['invoice'].currency_id), 'note_attrs': {}})
         return vals
 
+    def _get_amount_in_text(self, amount, currency):
```

Comment:
Just a nitpick: I would maybe change it to `_get_amount_in_text_note` or something like that, as you don't exclusively return the amount in text, since you add a specific "YALNIZ :" in the beginning.
Or add the "YALNIZ :" in the calling function and return only the amount in text here